### ISSUE TITLE
Preview style di fatturapa non è legato correttamente alla company

### ISSUE THREAD

**Issue by michelerusti** [OCA/l10n-italy#2965]
## Module
l10n_it_fatturapa

## Describe the bug
Quando si fa la stampa della preview della fattura, se sono presenti più company con templates diversi, non sempre viene preso il template corretto per via di come viene legato alla company in fase di lettura

## To Reproduce
**Affected versions**:
## To Reproduce
- [ ] 12.0: ?
- [x] 14.0: https://github.com/OCA/l10n-italy/pull/2964

1. Multi-company > impostazioni
2. Selezionare un template diverso per la preview della fatturapa
3. Contabilità > Encoming e-bill
4. Selezionarne una 
5. Stampare la preview

**Expected behavior**
Stampare il file con il template corretto

**Comment by TheMule71** [comment]
Per corretto intendi 
- rispetto alla company correntemente selezionata per l'utente o
- rispetto alla company della fattura?

**Comment by dcorio** [comment]
Non sono sicurissimo sia correlato a questo bug, ma i clienti si lamentano del fatto che ogni tanto si resetta il template del preview e se avevano selezionato quello di Assosoftware, gli viene proposto quello standard AdE.
Guardando il codice della PR, mi pare di capire che possa centrare.

Quindi il problema non riguarda solo i casi di multi-company

**Comment by michelerusti** [comment]
> Per corretto intendi
> 
>     * rispetto alla company correntemente selezionata per l'utente o
> 
>     * rispetto alla company della fattura?

Per company selezionata dall'utente visto che si può impostare un template diverso in base alla company su cui si decide di lavorare

**Comment by michelerusti** [comment]
@dcorio l'errore si presenta quando in res.company ci sono più righe, eventualmente prova a controllare quello!

**Comment by TheMule71** [comment]
Credo che il bug sia questo:
issue: https://github.com/odoo/odoo/pull/104284
fix: odoo/odoo@bfee55b14c918c44ebbb2563d5dc0e102edf396e

**Comment by michelerusti** [comment]
@TheMule71 
Si, nella 16.0 penso sia stato risolto con quel metodo aggiunto [quando si recupera le companies](https://github.com/odoo/odoo/blob/16.0/odoo/api.py#L668)
Sulla 14.0 però non è stato ancora portata la modifica ed è come prima, pensi che arriverà in futuro?

**Comment by SirAionTech** [comment]
Ho trovato questo problema in `16.0` mentre facevo revisione di https://github.com/OCA/l10n-italy/pull/3891:

[Untitled_ Feb 15, 2024 9_59 AM.webm](https://github.com/OCA/l10n-italy/assets/143788994/4df550dd-e48d-4bc2-afc6-d63a7acdfa0e)

Gli step sono:

1. Nell'azienda principale dell'utente seleziono un template diverso da Assosoftware
2. Cambio azienda (solo un'azienda selezionata)
3. Nelle impostazioni dell'azienda seleziono la preview AssoSoftware
4. Creo una fattura elettronica per fornitori
5. Visualizzo l'anteprima

Comportamento attuale:
L'anteprima è con un foglio di stile diverso da Assosoftware.
Più precisamente, il foglio di stile usato è quello dell'azienda principale dell'utente.

Comportamento atteso:
L'anteprima è di Assosoftware

@michelerusti @TheMule71 ditemi voi se è lo stesso problema descritto da questa issue perché dalla descrizione non lo capisco perché non si parla di cambiare azienda.
Se è un altro problema, apro un'altra issue.

**Comment by TheMule71** [comment]
Allora abbiamo discusso un po' in call.

per come la vedo io, lo stile di preview di default dovrebbe essere quello della azienda dell'allegato. Per cui né `env.user.company_id`, né `env.company`, ma `attachment.company_id`.

Se l'utente desidera modificare il modo in cui vede il preview, non ha molto senso che cerchi tra le aziende a cui è abilitato una che abbia il preview che gli piace, e selezioni quella come azienda corrente.  Considerate che l'utente "normale" non ha il controllo sulle impostazioni della company o sulle proprie company.   Vuol dire che da un giorno all'altro l'utente abituato al preview AssoSoftware potrebbe essere tagliato fuori se qualcun altro modifica l'impostazione delle company che era solito usare.

E in linea di pricipio è brutto usare la selezione della company corrente come modo per selezionare quella che è per logica un'impostazione di preferenza utente.

Per cui la proposta è:  mettere come stile di preview di un allegato di default quello della sua company, e aggiungere una preferenza a livello profilo utente, che possa (se specificato) fare l'override del default.   Anche dal punto di vista funzionale, utente va sul profilo e si cambia lo stile di preview, che è più ovvio. Se lo lascia vuoto, lo vede in base all'impostazione della company dell'allegato.

**Comment by primes2h** [comment]
> Allora abbiamo discusso un po' in call.
> 
> per come la vedo io, lo stile di preview di default dovrebbe essere quello della azienda dell'allegato. Per cui né `env.user.company_id`, né `env.company`, ma `attachment.company_id`.
> 
> Se l'utente desidera modificare il modo in cui vede il preview, non ha molto senso che cerchi tra le aziende a cui è abilitato una che abbia il preview che gli piace, e selezioni quella come azienda corrente. Considerate che l'utente "normale" non ha il controllo sulle impostazioni della company o sulle proprie company. Vuol dire che da un giorno all'altro l'utente abituato al preview AssoSoftware potrebbe essere tagliato fuori se qualcun altro modifica l'impostazione delle company che era solito usare.
> 
> E in linea di pricipio è brutto usare la selezione della company corrente come modo per selezionare quella che è per logica un'impostazione di preferenza utente.
> 
> Per cui la proposta è: mettere come stile di preview di un allegato di default quello della sua company, e aggiungere una preferenza a livello profilo utente, che possa (se specificato) fare l'override del default. Anche dal punto di vista funzionale, utente va sul profilo e si cambia lo stile di preview, che è più ovvio. Se lo lascia vuoto, lo vede in base all'impostazione della company dell'allegato.

Concordo, a tal riguardo riporto anche questa discussione (quando è stata introdotta la funzionalità):

https://github.com/OCA/l10n-italy/pull/1036#pullrequestreview-208469225
